R/fwrite.R

```diff
@@ -0,0 +1,62 @@
+fwrite <- function(dt, file.path, append = FALSE, quote = TRUE,
```

Comment:
2) Gotcha. Not sure how important the consistency is, but maybe just follow `write.table` to this end:

```
if (file == "") 
  file <- stdout()
else if (is.character(file)) {
  file <- if (nzchar(fileEncoding)) 
    file(file, ifelse(append, "a", "w"), encoding = fileEncoding)
  else file(file, ifelse(append, "a", "w"))
  on.exit(close(file))
}
else if (!isOpen(file, "w")) {
  open(file, "w")
  on.exit(close(file))
}
if (!inherits(file, "connection")) 
  stop("'file' must be a character string or connection")
```

And [here](https://github.com/wch/r-source/blob/b156e3a711967f58131e23c1b1dc1ea90e2f0c43/src/library/utils/src/io.c#L1038) is the C side of the code, to the extent that it helps